plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerDelegate.java

```diff
@@ -65,12 +68,27 @@ public IStatus canModifyModules(IModule[] add, IModule[] remove) {
           if (status != null && !status.isOK()) {
             return status;
           }
+          status = hasAppEngineStandardFacet(module);
+          if (status != null && !status.isOK()) {
+            return status;
+          }
         }
       }
     }
     return Status.OK_STATUS;
   }
 
+  private IStatus hasAppEngineStandardFacet(IModule module) {
+    try {
+      if (AppEngineStandardFacet.hasAppEngineFacet(ProjectFacetsManager.create(module.getProject()))) {
+        return Status.OK_STATUS;
+      }
+    } catch (CoreException ex) {
+      return StatusUtil.error(this, Messages.FACETED_PROJECT_CREATION_FAILED, ex);
```

Comment:
It's strange to have an error that a project could not be created from a predicate method.  Maybe _Not a faceted project_?